cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -975,12 +964,12 @@ TEST(GroupBy, SumMeanProductDecimal) {
                                },
                                {table->GetColumnByName("key")},
                                {
-                                   {"hash_sum", nullptr},
-                                   {"hash_sum", nullptr},
-                                   {"hash_mean", nullptr},
-                                   {"hash_mean", nullptr},
-                                   {"hash_product", nullptr},
-                                   {"hash_product", nullptr},
+                                   {"hash_sum", nullptr, "agg_0", "hash_sum"},
+                                   {"hash_sum", nullptr, "agg_1", "hash_sum"},
+                                   {"hash_mean", nullptr, "agg_2", "hash_mean"},
+                                   {"hash_mean", nullptr, "agg_3", "hash_mean"},
+                                   {"hash_product", nullptr, "agg_4", "hash_product"},
+                                   {"hash_product", nullptr, "agg_5", "hash_product"},
```

Comment:
I don't think we should have to make these changes and they do not make the test easier to read.  Could we change the definition of `GroupByTest` so that instead of taking in `const std::vector<internal::Aggregate>& aggregates` it takes in `const std::vector<TestAggregate>& aggregates` where we define:

```
struct TestAggregate {
  std::string function;
  const FunctionOptions* options;
};
```

The old style makes sense for these unit tests since we don't really care about what we are naming the columns.